File: src/C3.Catalogue/Brands/Brand.vb
Namespace Brands

    Public NotInheritable Class Brand

        Public Sub New(name As String, code As String, addedAt As DateTime, notes As String)
            Me.Name = name
            Me.Code = code
            Me.AddedAt = addedAt
            Me.Notes = notes
        End Sub

        Public ReadOnly Property Name As String
        Public ReadOnly Property Code As String
        Public ReadOnly Property AddedAt As DateTime
        Public ReadOnly Property Notes As String

    End Class

End Namespace

File: src/C3.Catalogue/Brands/Brand.cs
using System;

namespace C3.Catalogue.Brands
{
    public sealed class Brand
    {
        public Brand(string name, string code, DateTime addedAt, string notes)
        {
            Name = name;
            Code = code;
            AddedAt = addedAt;
            Notes = notes;
        }

        public string Name { get; }

        public string Code { get; }

        public DateTime AddedAt { get; }

        public string Notes { get; }
    }
}
